src/components/Devnet/Devnet.tsx

```diff
@@ -1,43 +1,52 @@
 import { useEffect, useState } from 'react';
 import { getInstance } from '../../fhevmjs';
-
 import './Devnet.css';
-import { Eip1193Provider, getAddress } from 'ethers';
+import { Eip1193Provider } from 'ethers';
+
+const toHexString = (bytes: Uint8Array) =>
+  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
 
-const toHexString = (bytes: Uint8Array) => bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
+export type DevnetProps = {
+  account: string;
+  provider: Eip1193Provider;
+};
 
-export type DevnetProps = { account: string; provider: Eip1193Provider };
+const CONTRACT_ADDRESS = '0x309cf2aae85ad8a1db70ca88cfd4225bf17a7482';
 
 export const Devnet = ({ account, provider }: DevnetProps) => {
   const [handles, setHandles] = useState<Uint8Array[]>([]);
   const [encryption, setEncryption] = useState<Uint8Array>();
-  const [eip712, setEip712] = useState<ReturnType<typeof instance.createEIP712>>();
+  const [eip712, setEip712] =
+    useState<ReturnType<typeof instance.createEIP712>>();
   const instance = getInstance();
 
+  // Handle EIP712 setup
   useEffect(() => {
     const { publicKey } = instance.generateKeypair();
-    const eip = instance.createEIP712(publicKey, '0x309cf2aae85ad8a1db70ca88cfd4225bf17a7482');
+    const eip = instance.createEIP712(publicKey, CONTRACT_ADDRESS);
     setEip712(eip);
   }, [instance]);
 
-  const encrypt = (val: number) => {
+  const encrypt = async (val: number) => {
+    if (!account) {
```

Comment:
Useless because this is handled by a component above. account is a mandatory props so you're always connected when you see this component.

Reply:
done in new commit i just pushed